edge/stack/stack.go

```diff
@@ -196,6 +198,19 @@ func (manager *StackManager) Start() error {
 				ctx := context.TODO()
 
 				if stack.Action == actionDeploy || stack.Action == actionUpdate {
+					for stackImage, cacheImage := range stack.ImageMapping {
+						//pull cache image
+						//push stackImage to docker
+						// TODO (on server too): this code doesn't like image tag == "<none>", so we need to default that to ":latest"
+						// TODO: make sure the image is right for this agent's arch.
+						// see `skopeo inspect --raw docker://nginx | jq`
+						// TODO: might need to be able to retry this, as the server may not have finished syncing
+						err := copyImage(ctx, cacheImage, stackImage)
+						if err != nil {
+							log.Println("[ERROR] copying image %s -> %s: %s", cacheImage, stackImage, err.Error())
```

Comment:
I'm thinking how could we handle this type of errors, won't the `deployStack` fail if something fails here?